creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -223,66 +217,76 @@ class SnapbackSM {
-        )
-      }
-
-      try {
-        await redis.set('stateMachineQueueLatestJobStart', Date.now())
-        await this.processStateMachineOperation(jobId)
-        await redis.set('stateMachineQueueLatestJobSuccess', Date.now())
-      } catch (e) {
-        this.logError(
-          `StateMachineQueue: Processing error on jobId ${jobId}: ${e}`
-        )
-      }
-
-      return {}
-    })
+    if (!this.inittedJobProcessors) {
+      // Initialize stateMachineQueue job processor (aka consumer)
+      this.stateMachineQueue.process(1 /** concurrency */, async (job) => {
+        this.log('StateMachineQueue: Consuming new job...')
+        const { id: jobId } = job
 
-    // Initialize manualSyncQueue job processor
-    this.manualSyncQueue.process(
-      this.MaxManualRequestSyncJobConcurrency,
-      async (job) => {
         try {
-          await this.processSyncOperation(job, SyncType.Manual)
+          this.log(
+            `StateMachineQueue: New job details: jobId=${jobId}, job=${JSON.stringify(
+              job
+            )}`
+          )
         } catch (e) {
-          this.logError(`ManualSyncQueue processing error: ${e}`)
+          this.logError(
+            `StateMachineQueue: Failed to log details for jobId=${jobId}: ${e}`
+          )
         }
-      }
-    )
 
-    // Initialize recurringSyncQueue job processor
-    this.recurringSyncQueue.process(
-      this.MaxRecurringRequestSyncJobConcurrency,
-      async (job) => {
         try {
-          await this.processSyncOperation(job, SyncType.Recurring)
+          await redis.set('stateMachineQueueLatestJobStart', Date.now())
+          await this.processStateMachineOperation(jobId)
+          await redis.set('stateMachineQueueLatestJobSuccess', Date.now())
         } catch (e) {
-          this.logError(`RecurringSyncQueue processing error ${e}`)
+          this.logError(
+            `StateMachineQueue: Processing error on jobId ${jobId}: ${e}`
+          )
         }
-      }
-    )
+
+        return {}
+      })
+
+      // Initialize manualSyncQueue job processor
+      this.manualSyncQueue.process(
+        this.MaxManualRequestSyncJobConcurrency,
+        async (job) => {
+          try {
+            await this.processSyncOperation(job, SyncType.Manual)
+          } catch (e) {
+            this.logError(`ManualSyncQueue processing error: ${e}`)
+          }
+        }
+      )
+
+      // Initialize recurringSyncQueue job processor
+      this.recurringSyncQueue.process(
+        this.MaxRecurringRequestSyncJobConcurrency,
+        async (job) => {
+          try {
+            await this.processSyncOperation(job, SyncType.Recurring)
+          } catch (e) {
+            this.logError(`RecurringSyncQueue processing error ${e}`)
+          }
+        }
+      )
+      this.inittedJobProcessors = true
+    }
 
     // Start at a random userId to avoid biased processing of early users
     const latestUserId = await this.getLatestUserId()
```

Comment:
this could also cause the `_initSnapbackSM Error Error: No discovery provider currently selected, exiting` error and subsequent retries of `Queue#process` above